gateway/ln-gateway/src/lib.rs

```diff
@@ -670,12 +672,24 @@ impl Gateway {
             address,
             federation_id,
         } = payload;
-
         let client = self.select_client(federation_id).await?;
-        // TODO: This should probably be passed in as a parameter
-        let fees = client.get_withdraw_fee(address.clone(), amount).await?;
 
-        let operation_id = client.withdraw(address, amount, fees).await?;
+        // TODO: Fees should probably be passed in as a parameter
+        let (amount, fee) = match amount {
+            // If the amount is "all", then we need to subtract the fees from
+            // the amount we are withdrawing
+            BitcoinAmountOrAll::All => {
+                let balance = bitcoin::Amount::from_sat(client.get_balance().await.msats * 1000);
+                let fees = client.get_withdraw_fee(address.clone(), balance).await?;
+                (balance - fees.amount(), fees)
```

Comment:
You should do this in a loop imo to check that the fee you have to pay stayed the same. There is no guarantee that the fee cannot rise from withdrawing a different amount (maybe there is currently due to our UTXO selection approach, but I wouldn't count on it).

Reply:
Oh, if the fee is supposed to be both of Fedimint tx and on chain fees, then we should use msats for `balance` (and the `fee`) BTW, just as I thought though.

Also - the loop termination condition should probably be "not increasing" or something. I suspect "stayed the same" can lead to hangs.